net/captivedetection/captivedetection.go

```diff
@@ -23,6 +24,7 @@ import (
 
 // Detector checks whether the system is behind a captive portal.
 type Detector struct {
+	nower func() time.Time
```

Comment:
`clock func() time.Time` I think we often do?

`clock.Now()` looks nice.

and paves the way to using tstest.Clock later with the same name if/when needed